### ISSUE TITLE
timeout issue from __ensure_connected with 1000 workers

### ISSUE THREAD

**Issue by anishkej** [dask/dask-jobqueue#316]
I'm having some problems with trying to submit 1000 workers, but it works fine with 500 workers.  Not sure what's going on.

My code:
```
cluster = SGECluster(cores=1,
                         processes=1,
                         memory=4,
                         walltime='99999:0:0',
                         death_timeout=10000,
                         local_directory=/temp,
                         )
cluster.scale(num_workers)
client = dask.distributed.Client(cluster)

```
the error:
```
    client = dask.distributed.Client(cluster)
  File "/anaconda3/lib/python3.6/site-packages/distributed/client.py",
 line 712, in __init__
    self.start(timeout=timeout)
  File "/anaconda3/lib/python3.6/site-packages/distributed/client.py",
 line 858, in start
    sync(self.loop, self._start, **kwargs)
  File "/anaconda3/lib/python3.6/site-packages/distributed/utils.py",
line 331, in sync
    six.reraise(*error[0])
  File "/anaconda3/lib/python3.6/site-packages/six.py", line 693, in reraise
    raise value
  File "/anaconda3/lib/python3.6/site-packages/distributed/utils.py", line 316, in f
    result[0] = yield future
  File "/anaconda3/lib/python3.6/site-packages/tornado/gen.py", line 729, in run
DEBUG:dask_jobqueue.core:started job: 735
    value = future.result()
  File "/anaconda3/lib/python3.6/site-packages/tornado/gen.py", line 736, in run
    yielded = self.gen.throw(*exc_info)  # type: ignore
  File "/anaconda3/lib/python3.6/site-packages/distributed/client.py", line 954, in _start
    yield self._ensure_connected(timeout=timeout)
  File "/anaconda3/lib/python3.6/site-packages/tornado/gen.py", line 729, in run
    value = future.result()
  File "/anaconda3/lib/python3.6/site-packages/tornado/gen.py", line 736, in run
    yielded = self.gen.throw(*exc_info)  # type: ignore
  File "/anaconda3/lib/python3.6/site-packages/distributed/client.py", line 1015, in _ensure_connected

/anaconda3/bin/python -m distributed.cli.dask_worker tcp://x.x.x.x:33137 --nthreads 1 --memory-limit 4.00GB --name dask-worker--${JOB_ID}-- --death-timeout 10000 --local-directory /temp

    timedelta(seconds=timeout), self._update_scheduler_info()
  File "/anaconda3/lib/python3.6/site-packages/tornado/gen.py", line 729, in run
    value = future.result()
tornado.util.TimeoutError: Timeout
```

thanks!

**Comment by guillaumeeb** [comment]
Hi @anishkej,

It looks like it comes from your client taking to much time to respond to the task submission.

When looking at the dashboard, do you see 1000 workers up and running? Maybe they take time to all connect to the scheduler, which makes the client unresponsive.

Maybe you should use `client.wait_for_workers` (or maybe the function has another name, but something like this), or increase the timeout on the Client side.

**Comment by anishkej** [comment]
Great question.  No, not all 1000 workers up and running right away.  I'm running this on AWS, and using an auto-scaling group via AWS ParallelCluster.  I should have mentioned that, because this means all my worker instances may not be available right way.

currently I submit my tasks via:
```
tasks = [client.submit(stuff, *args) for args in arguments]
```

and then I process the tasks as they complete via:
```
for task in dask.distributed.as_completed(tasks):
        logging.info('Complete: {0}'.format(task))

```
I don't see a way to wait_for_workers type concept.  Am I missing something? Would it conceptually work like the following:
```
while ((client.status == "running") and (len(client.scheduler_info()["workers"]) < nworkers)):
    sleep(1.0)
```
I found that elsewhere, but that didn't work for me either.also tried the follow (that I found elsewhere), but that didn't do anything.

Any idea on how to work around the fact that not all my workers are available, and that I might not even get all my workers from AWS?
